pset-functions.js

```diff
@@ -208,6 +343,41 @@ console.log('Problem 5 - bonus:')
 
 // Add your code below this line
 
+
+function wordCount(phrase) {
+
+  // making sure we're dealing with a string
+  if (typeof phrase === "string") {
+    phrase = phrase.toLowerCase()
+  } else {
+    return "wordCount() requires a string parameter."
+  }
+
+  // clean up string by removing punctuation
+  phrase = phrase.replace(/[.,\/#!$%\^&\*;:{}=\-_`~()]/g,"")
+
+  // break up single string
+  const wordArray = phrase.split(' ')
+  // create array to count occurrences
+  const countArray = {}
+
+  // running through array and counting occurences
+  for (let i = 0; i < wordArray.length; i++) {
+    word = wordArray[i]
+    if (countArray[word] === undefined) {
+      countArray[word] = 1
+    } else {
+      countArray[word] += 1
+    }
+  }
+  return countArray
+}
```

Comment:
Nate, I love how you used an object to store your updated word count; great use of a for loop mixed with conditional if/else statements. I notice you are removing punctation with is another great example of taking potential edge cases into consideration.

Overall, you did a really amazing job on this homework! All edge cases considered, very clean formatted code, and great logic to finding solutions. Keep up the great work!